client/js/components/procedure/StatementSegmentsList/StatementSegmentsList.vue

```diff
@@ -580,6 +582,20 @@ export default {
       this.currentAction = action || defaultAction
     },
 
+    /**
+     * This prevents the user from unintentionally deleting an unsaved text by synchronizing the local
+     * statement in StatementMeta.vue (which also emits the local statement when saving only metadata)
+     * with the statements from store. The editor automatically updates the state of statements in the
+     * store when registering an input. This only occurs when a statement has not been segmented already.
+     *
+     * @param {object} statement - The local statement of StatementMeta.vue.
+     */
+    synchronizeFullText (statement) {
+      if (statement.attributes.fullText !== this.statement.attributes.fullText) {
```

Comment:
After looking into it a little deeper: This means that when clicking the meta data save button, the statement text is also saved? What was the reasoning behind choosing this solution?

Reply:
Because the view contains both scenarios if a statement has segments and if not (the editor only appearing if the statement has no segments). To handle this, the StatementMeta form and the rest was decoupled. Also there is a check to minimalize the request payload by checking if a value has changed and only data which needs to be updated will be part of the request payload.

The problem now is that the `StatementMeta.vue` determines which data needs to be in the request payload by a `local statement ` data prop which contains only initial data. Like that, the store state of `statements` will never contain the updated text field but will reset unsaved changes due to a store reset action within `saveStatements`.

I did not want to refactor this whole view or break all the logic which I find useful indeed. Hence I tried to solve the main problem of a resetted unsaved text but keep all the rest still functional. This is also a candidate for a bigger refactoring imho.

Reply:
> This is also a candidate for a bigger refactoring imho.

Agreed!

For now, would it maybe make sense to use a `dpconfirm` to let the user decide if they want to save the text, too, or not? So if they cancel, nothing is saved and they can reset the statement text if desired before saving the meta data. I don't know if I'm overcomplicating things, but as a user, I would not expect the text to be saved with the meta data save button, so I think we should somehow let them know.

Reply:
This is a great idea, this reduces also the possible confusion a user maybe experience with those multiple save buttons in one view.

Reply:
Done here (finally): https://github.com/demos-europe/demosplan-core/pull/1049/commits/0582115898dae3a08cef28dc0e889f1976d4a908

Reply:
Isn't it the statement text instead of the recommendation text? I'd suggest something like "Sie haben Änderungen am Text der Stellungnahme vorgenommen. Wenn Sie fortfahren, werden diese ebenfalls gespeichert."

Reply:
True, I confused myself because the browser tab displayed "Erwiderungen verfassen" , [a493c54](https://github.com/demos-europe/demosplan-core/pull/1049/commits/a493c543c39cb64227ec4c24f505c74479846c24)